src/Schema/Table.php

```diff
@@ -41,18 +44,27 @@ class Table extends AbstractAsset
     /** @var SchemaConfig|null */
     protected $_schemaConfig = null;
 
+    /** @var Index[] */
+    private $implicitIndexes = [];
+
     /**
      * @param string                 $tableName
      * @param Column[]               $columns
      * @param Index[]                $indexes
+     * @param UniqueConstraint[]     $uniqueConstraints
      * @param ForeignKeyConstraint[] $fkConstraints
-     * @param int                    $idGeneratorType
      * @param mixed[]                $options
      *
      * @throws DBALException
      */
-    public function __construct($tableName, array $columns = [], array $indexes = [], array $fkConstraints = [], $idGeneratorType = 0, array $options = [])
-    {
+    public function __construct(
+        $tableName,
```

Comment:
Should this be `string $name`?

Reply:
This might break somewhere else. No worth doing it here as part of this PR.

Reply:
Let's agree that the starter of the discussion resolves it. The very first instruction of the method checks the length of the table name. What else than a string can it be to work? Do you expect consumers to pass an object that implements `__toString()`?

Reply:
I've ran a test and it works ok on both DBAL and ORM.
Personally, my changes have no correlation to the parameter $name in Table class, and this discussion was outside of scope of this PR (which leads to longer cycle time). But anyway, I've done the change as requested.